### PR TITLE
Remove gen-collateral when rebuilding

### PR DESCRIPTION
**Pull request by joey0320**
**Related PRs / Issues**:

This PR addresses #1341.
Removes $(OUT_DIR) which is `gen-collateral` when re-compiling.

**Type of change**:
- [x] Bug fix
- [ ] New feature
- [ ] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [x] Build system change
- [ ] Other

**Contributor Checklist**:
- [x] Did you set `main` as the base branch?
- [x] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [x] Did you state the type-of-change/impact?
- [x] Did you delete any extraneous prints/debugging code?
- [x] Did you mark the PR with a `changelog:` label?
- [ ] (If applicable) Did you update the conda `.conda-lock.yml` file if you updated the conda requirements file?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you add a test demonstrating the PR?

- [ ] (If applicable) Did you mark the PR as `Please Backport`?

### CODE REVIEW

**Review by abejgonzalez — CHANGES_REQUESTED** [review]
Why not just change the FIRTOOL recipe to `rm -rf $(OUT_DIR)` and list it as a target? That seems less invasive.

**Comment by jerryz123** [comment]
In the process, can we also rename `OUT_DIR` to something more descriptive? `GEN_COLLATERAL_DIR`?

**Review by jerryz123 — APPROVED** [review]
Implementation looks clean. I'll test this once this gets merged.

**Comment by abejgonzalez** [comment]
The prior approach didn't enforce that the collateral directory would be rebuilt. This new approach basically forces `sim_files` to be generated after the FIRTOOL target (this was already done with the `sim_common_files` target) and removes the collateral directory in the FIRTOOL target.

File: vlsi/Makefile
Comment (jerryz123) [inline vlsi/Makefile:67]: Should this be here?

Comment (harrisonliew) [reply vlsi/Makefile:67]: No, also this would collide with #1324.

Comment (abejgonzalez) [reply vlsi/Makefile:67]: pre-commit will block committing without this so I just added it in. I'll remove it.

**Review by abejgonzalez — APPROVED** [review]
The renaming LGTM. Pending reviews from others on the `rm -rf`

**Review by jerryz123 — APPROVED** [review]

**Pull request merged**
